src/main/java/org/jetbrains/research/anticopypaster/ide/ui/CustomMetricsMenuAction.java

```diff
@@ -0,0 +1,89 @@
+package org.jetbrains.research.anticopypaster.ide.ui;
+
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
+
+import java.io.FileWriter;
+import java.io.IOException;
+
+
+public class CustomMetricsMenuAction extends AnAction {
+
+    private static final String FILE_PATH = "./custom_metrics.txt";
+
+    public void actionPerformed(AnActionEvent e) {
+        CustomMetricsMenu dialog = new CustomMetricsMenu();
+        if (dialog.showAndGet()) {
+            // user clicked "OK", retrieve the selected values from the dropdowns
+            String keywordsDropdownValue = dialog.getKeywordsDropdownValue();
+            Boolean keywordsCheckboxValue = dialog.getKeywordsCheckboxValue();
+            String couplingDropdownValue = dialog.getCouplingDropdownValue();
+            Boolean couplingCheckboxValue = dialog.getCouplingCheckboxValue();
+            String sizeDropdownValue = dialog.getSizeDropdownValue();
+            Boolean sizeCheckboxValue = dialog.getSizeCheckboxValue();
+            String complexityDropdownValue = dialog.getComplexityDropdownValue();
+            Boolean complexityCheckboxValue = dialog.getComplexityCheckboxValue();
+
+            // create new customMetricsModel object to return to the backend
+            CustomMetricsModel customMetricsModel = new CustomMetricsModel();
+            customMetricsModel.keywordsDropdownValue = keywordsDropdownValue;
+            customMetricsModel.keywordsCheckboxValue = String.valueOf(keywordsCheckboxValue);
+            customMetricsModel.couplingDropdownValue = couplingDropdownValue;
+            customMetricsModel.couplingCheckboxValue = String.valueOf(couplingCheckboxValue);
+            customMetricsModel.sizeDropdownValue = sizeDropdownValue;
+            customMetricsModel.sizeCheckboxValue = String.valueOf(sizeCheckboxValue);
+            customMetricsModel.complexityDropdownValue = complexityDropdownValue;
+            customMetricsModel.complexityCheckboxValue = String.valueOf(complexityCheckboxValue);
+
+            // Save the values to the file
+            dialog.loadState(customMetricsModel);
+
+
+            System.out.println("---------------------------------------------------------");
```

Comment:
Might be good to get rid of this debug stuff, but also doesn't really matter in the long run.